### PR TITLE
Http tests

### PR DESCRIPTION
**Pull request by opetkevich**
This change is [Reviewable](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/1855)

### CODE REVIEW

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (c045007)

**Comment by ThaliTester** [comment]
Test 115905227 (c045007) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (c045007)

See https://github.com/ThaliTester/TestResults/tree/115905227c045007_Http_tests_opetkevich/ for the logs

**Comment by yaronyg** [comment]
Review status: 0 of 1 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 6 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwnPv5tJyERjfopfed:-KhwnPv5tJyERjfopfee:brp6i82) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L6)):*
> ```JavaScript
> var tape = require('../lib/thaliTape');
> var Promise = require('lie');
> var EE = require('events').EventEmitter;
> ```

NB: It wouldn't hurt to make this a little clearer, say EventEmitter. EE is kind of obscure.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 25 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwmJhcBSrO_cyUkGkh:-KhwmJhcBSrO_cyUkGki:b-b0uzfo) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L25)):*
> ```JavaScript
> });
> 
> function listen(server, port) {
> ```

NB: The name is a bit opaque. How about something like serverIsListening()?

---

*[test/www/jxcore/bv_tests/testHttp.js, line 53 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Khwme1nPeQPwP6hw0aK:-Khwme1nPeQPwP6hw0aL:bc61owc) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L53)):*
> ```JavaScript
>       response.on('end', function () {
>         response.removeListener('error', reject);
>         var data = Buffer.concat(received);
> ```

NB: We have two variables named data here and they are being used for two different purposes. I would suggest renaming them. Say requestData and responseData.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 120 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Khwo92o67vwblY8ESJi:-Khwo92o67vwblY8ESJj:b-tsdknp) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L120)):*
> ```JavaScript
>   }
> 
>   function find(numberOfPeers) {
> ```

Shouldn't this really be a proper class and not a function with with another function (cleanup) on the side? I think prototype is crying out to be used here.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 121 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Khwnnk7dEMIbuAD-ItE:-Khwnnk7dEMIbuAD-ItF:bh6bicp) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L121)):*
> ```JavaScript
> 
>   function find(numberOfPeers) {
>     var keys = Object.keys(peers);
> ```

NB: Personally I would have just passed in peers as an argument. It makes the function completely stand alone which is always nice from a debugging and readability perspective.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 149 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwpVqShkj4y5K0D_n6:-KhwpVqShkj4y5K0D_n7:b-fpomdb) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L149)):*
> ```JavaScript
> };
> 
> test('Single coordinated request ios native',
> ```

I'm surprised to not see any kind of connection failure testing.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 171 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwoPJpAhAJVAPPpZ4j:-KhwoPJpAhAJVAPPpZ4k:b-bqju2t) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L171)):*
> ```JavaScript
> 
>     Promise.all([ listenNative, listenServer ])
>       .then(function (r) {
> ```

NB: r is a bit obscure. It's especially hard to follow after a Promise.all where there are going to be two responses. I would use something a bit more verbose like allResponses.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 203 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwpE2t_oTU_LhIW1lw:-KhwpE2t_oTU_LhIW1lx:b-ipdi6i) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L203)):*
> ```JavaScript
>   });
> 
> test('Multiple coordinated request ios native',
> ```

I would suggest having different sizes of data for different requests. Some using small data and some using large (e.g. more than 2 or 3 k). This will stress the system a bit more.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by yaronyg** [comment]
Please respond to comments. Thanks!

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-:-KhxFztwwmgg_UQLLnTV:b-m5vgd3)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (08f3133)

**Comment by ThaliTester** [comment]
Test 115905227 (08f3133) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (08f3133)

See https://github.com/ThaliTester/TestResults/tree/11590522708f3133_Http_tests_opetkevich/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (7abb2e7)

**Comment by ThaliTester** [comment]
Test 115905227 (7abb2e7) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (7abb2e7)

See https://github.com/ThaliTester/TestResults/tree/1159052277abb2e7_Http_tests_opetkevich/ for the logs

**Comment by opetkevich** [comment]
@yaronyg test('Single coordinated request ios native'
I'm surprised to not see any kind of connection failure testing.
Could you please clarify this comment with more details?

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (46edbca)

**Comment by ThaliTester** [comment]
Test 115905227 (46edbca) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (46edbca)

See https://github.com/ThaliTester/TestResults/tree/11590522746edbca_Http_tests_opetkevich/ for the logs

**Comment by yaronyg** [comment]
@opetkevich In the future please put questions in reviewable. It makes things much easier to track. But in answer to your question I was surprised not to see tests where we intentionally cause the HTTP client and/or server connection to fail and see that everything is still working. In other words, connect, then have the client/or server disconnect and then connect again. We actually already have tests like that but we also have tests that do what the iOS only tests do so if the existing tests didn't work for successful connections then they wouldn't work for unsuccessful connections either.

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (d1d7212)

**Comment by ThaliTester** [comment]
Test 115905227 (d1d7212) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (d1d7212)

See https://github.com/ThaliTester/TestResults/tree/115905227d1d7212_Http_tests_opetkevich/ for the logs

**Comment by opetkevich** [comment]
Review status: 0 of 3 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 6 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwnPv5tJyERjfopfed:-Ki4gXcjQSwUK9tHvzuW:b-896fix) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L6)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

NB: It wouldn't hurt to make this a little clearer, say EventEmitter. EE is kind of obscure.
</blockquote></details>

Done.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 25 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwmJhcBSrO_cyUkGkh:-Ki4gQM9pZ_e57XsLKda:b-896fix) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L25)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

NB: The name is a bit opaque. How about something like serverIsListening()?
</blockquote></details>

Done.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 53 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Khwme1nPeQPwP6hw0aK:-Ki4gWt9a418qll8xkC_:b-896fix) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L53)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

NB: We have two variables named data here and they are being used for two different purposes. I would suggest renaming them. Say requestData and responseData.
</blockquote></details>

Done.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 203 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwpE2t_oTU_LhIW1lw:-Ki4gbgQVm3VsNLza8H4:b-896fix) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L203)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

I would suggest having different sizes of data for different requests. Some using small data and some using large (e.g. more than 2 or 3 k). This will stress the system a bit more.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by yaronyg** [comment]
Reviewed 2 of 2 files at r2, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 133 at r4](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6Au1Y3PPWDU-DVp1c:-Ki6Au1Zi78z3760PswR:bz8n9p9) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/46edbca684589a852ff34e4996aab77cd5ffec4d/test/www/jxcore/bv_tests/testThaliMobileNative.js#L133)):*
> ```JavaScript
> });
> 
> function pad(s) { return ('0'+s).slice(-2); }
> ```

This function doesn't make sense to me. Why would a function call pad created a new array starting with 0 and then just return the last 2 characters?

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 134 at r5](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6B7OUbx65A1pZwDRZ:-Ki6B7OVMON0_vz9dt3M:bq37h7v) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d1d7212ff07faf8467426cd8e7103dcaaea49c12/test/www/jxcore/bv_tests/testThaliMobileNative.js#L134)):*
> ```JavaScript
> 
> function pad(s) { return ('0'+s).slice(-2); }
> function pb(b) {
> ```

Both the function name and the argument name are obscure. Please, with the exception of iterator counters we shouldn't have single letter variables anywhere.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 137 at r5](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6BzAgsamd6eEKBzHQ:-Ki6BzAholy716WQhwJ7:b-o14c17) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d1d7212ff07faf8467426cd8e7103dcaaea49c12/test/www/jxcore/bv_tests/testThaliMobileNative.js#L137)):*
> ```JavaScript
>   var result = 'Buffer <';
>   var l = b.length;
>   for (var i = 0; i < l; i++) {
> ```

This logic requires me to think to understand it which means it's obscure. Please write it so I don't have to actually engage my brain to understand it. And no, this does not mean putting in comments. I suspect starting with good names would help.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 150 at r5](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6CNep0xlcMX85e6MY:-Ki6CNep0xlcMX85e6MZ:bm66liy) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d1d7212ff07faf8467426cd8e7103dcaaea49c12/test/www/jxcore/bv_tests/testThaliMobileNative.js#L150)):*
> ```JavaScript
>   var RECV = reverse ? '→' : '←';
>   var server = net.createServer(function (incomingSocket) {
>     log(f('received incoming connection'));
> ```

Why not move log inside of the (soon to be renamed) f function?

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 192 at r5](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6CTogMxaNpFfqK5i2:-Ki6CTogMxaNpFfqK5i3:bdn8mp9) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d1d7212ff07faf8467426cd8e7103dcaaea49c12/test/www/jxcore/bv_tests/testThaliMobileNative.js#L192)):*
> ```JavaScript
> }
> 
> // test.only('simple', function (t) {
> ```

Why is this code still here?

---

*[thali/NextGeneration/thaliMobileNativeWrapper.js, line 561 at r5](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki68ru_dUzRc7Xrf_Nx:-Ki68rubx0VzbLqZNsnV:b-vllo84) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d1d7212ff07faf8467426cd8e7103dcaaea49c12/thali/NextGeneration/thaliMobileNativeWrapper.js#L561)):*
> ```JavaScript
>  * | Call Start! | The object is not in start state. |
>  *
>  * @public
> ```

Please change the JSDoc to specify that port is an integer and that it is optional. Please also add a note that it is only intended for testing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by opetkevich** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 133 at r4](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6Au1Y3PPWDU-DVp1c:-Ki6XzurpmauA_-j5TRY:bgct9n) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/46edbca684589a852ff34e4996aab77cd5ffec4d/test/www/jxcore/bv_tests/testThaliMobileNative.js#L133)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

This function doesn't make sense to me. Why would a function call pad created a new array starting with 0 and then just return the last 2 characters?
</blockquote></details>

@yaronyg 
I am really sorry for confusing you.
This is a result of a merge and this branch is not ready for a full review. I'll clean up the code, run eslint and update the jsDoc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by opetkevich** [comment]
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 149 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwpVqShkj4y5K0D_n6:-Ki6ZK2Cgshx3tI5eZwQ:b-11elcg) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L149)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

I'm surprised to not see any kind of connection failure testing.
</blockquote></details>

@yaronyg
 I have seen your previous comment regarding this issue.
We have added analogue tests for the HTTP because we thought that the issues we had with data transfer ("shifting data" on iOS) could be connected with the HTTPS.
We already have connection-check tests for HTTPS.
Do we need to recreate this functionality for HTTP tests? Though, if we need it maybe we can create an improvement issue for that? Also we need an update on the priority of the issue.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by yaronyg** [comment]
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 149 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KhwpVqShkj4y5K0D_n6:-Ki7KO0NJDKXEc3Zm4fp:b-45dnjs) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/c045007e71fdc3965540a446191adde6874459d7/test/www/jxcore/bv_tests/testHttp.js#L149)):*
<details><summary><i>Previously, opetkevich (Olga Petkevich) wrote…</i></summary><blockquote>

@yaronyg
 I have seen your previous comment regarding this issue.
We have added analogue tests for the HTTP because we thought that the issues we had with data transfer ("shifting data" on iOS) could be connected with the HTTPS.
We already have connection-check tests for HTTPS.
Do we need to recreate this functionality for HTTP tests? Though, if we need it maybe we can create an improvement issue for that? Also we need an update on the priority of the issue.
</blockquote></details>

O.k. I get it now. Makes sense.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 133 at r4](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-Ki6Au1Y3PPWDU-DVp1c:-Ki7KVVNHSnCbXP5o6pc:b95xeov) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/46edbca684589a852ff34e4996aab77cd5ffec4d/test/www/jxcore/bv_tests/testThaliMobileNative.js#L133)):*
<details><summary><i>Previously, opetkevich (Olga Petkevich) wrote…</i></summary><blockquote>

@yaronyg 
I am really sorry for confusing you.
This is a result of a merge and this branch is not ready for a full review. I'll clean up the code, run eslint and update the jsDoc.
</blockquote></details>

I actually figured something was wrong but I was marked as being asked for review. I'll wait until you are ready, just let me know.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by ThaliTester** [comment]
Test 115905227 (928b324) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (928b324)

See https://github.com/ThaliTester/TestResults/tree/115905227928b324_Http_tests_opetkevich/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (aeda9bc)

**Comment by ThaliTester** [comment]
Test 115905227 (aeda9bc) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (aeda9bc)

See https://github.com/ThaliTester/TestResults/tree/115905227aeda9bc_Http_tests_opetkevich/ for the logs

**Comment by dersim-davaod** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 361 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KiVyJl6-DreuNxBMCWP:-KiVyJl6-DreuNxBMCWQ:b9j0yky) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/aeda9bc19fbb3db1b55c78d84aa07db8f56023d2/test/www/jxcore/bv_tests/testThaliMobileNative.js#L361)):*
> ```JavaScript
> }
> 
> function wairForEvent(emitter, event) {
> ```

NB: rename wairForEvent to waitForEvent

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by yaronyg** [comment]
Reviewed 1 of 1 files at r7.
Review status: 1 of 3 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 123 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KiVz8Js4qrRDp_BJOFY:-KiVytqmNTuxBMeqAIbR:bq2k6nt) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/aeda9bc19fbb3db1b55c78d84aa07db8f56023d2/test/www/jxcore/bv_tests/testHttp.js#L123)):*
> ```JavaScript
> PeerFinder.prototype._setUpEvents = function() {
>   var self = this;
>   self._nonTCPPeerAvailabilityHandler = self._nonTCPPeerAvailabilityHandler
> ```

If you need to bind then you should do it to a different function. It's bad mojo to force an object level object to bind to a particular context.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (54538f7)

**Comment by ThaliTester** [comment]
Test 115905227 (54538f7) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (54538f7)

See https://github.com/ThaliTester/TestResults/tree/11590522754538f7_Http_tests_opetkevich/ for the logs

**Comment by yaronyg** [comment]
Please resolve the two remaining comments and then :lgtm: 

---

Reviewed 2 of 3 files at r6, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 37 at r8](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KjAYY0tWaP4c7xAyIsM:-KjAYY0uKYkg2WcFIvxU:b-63l9qh) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/54538f7a21d365242e2d7953377d2128d8d9e68d/test/www/jxcore/bv_tests/testHttp.js#L37)):*
> ```JavaScript
>         });
>       })
>       .catch(t.end)
> ```

If we get a catch don't we need to call t.fail and then t.end?

---

*[test/www/jxcore/bv_tests/testHttp.js, line 235 at r8](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KjAYzsPPrLsKNo8PeqO:-KjAYzsPPrLsKNo8PeqP:bot7te0) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/54538f7a21d365242e2d7953377d2128d8d9e68d/test/www/jxcore/bv_tests/testHttp.js#L235)):*
> ```JavaScript
>   function (t) {
>     var str = 'dummy str';
>     var firstReply = str.repeat(2000);
> ```

I don't mind but do you realize that where as the old line (randomString.generate(2000)) would have generated a 2000 character string that this one will generate a 2000 copies of 'dummy str' or a 9*2000 = 18,000 character string? Again, not a big deal. But I Just wanted to check that this was intentional.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 361 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KiVyJl6-DreuNxBMCWP:-KjAY-YFB05HCw1teQj9:b-9m7lau) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/aeda9bc19fbb3db1b55c78d84aa07db8f56023d2/test/www/jxcore/bv_tests/testThaliMobileNative.js#L361)):*
<details><summary><i>Previously, dersim-davaod wrote…</i></summary><blockquote>

NB: rename wairForEvent to waitForEvent
</blockquote></details>

This still hasn't been fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-:-KjAZQ8xIEgLV39u7vxS:b-mvcqq4)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (ef35344)

**Comment by ThaliTester** [comment]
Test 115905227 (ef35344) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (ef35344)

See https://github.com/ThaliTester/TestResults/tree/115905227ef35344_Http_tests_opetkevich/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (6e08aea)

**Comment by ThaliTester** [comment]
Test 115905227 (6e08aea) build started.

**Comment by dersim-davaod** [comment]
Review status: 1 of 3 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 37 at r8](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KjAYY0tWaP4c7xAyIsM:-KjDuvLc3_coHkiyDzbm:bujx8d8) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/54538f7a21d365242e2d7953377d2128d8d9e68d/test/www/jxcore/bv_tests/testHttp.js#L37)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

If we get a catch don't we need to call t.fail and then t.end?
</blockquote></details>

Yes, it was made by mistake (copy-paste probably). Fixed in the latest implementation.

---

*[test/www/jxcore/bv_tests/testHttp.js, line 235 at r8](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KjAYzsPPrLsKNo8PeqO:-KjE16OUkp800jhbAZ5q:b2fam5k) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/54538f7a21d365242e2d7953377d2128d8d9e68d/test/www/jxcore/bv_tests/testHttp.js#L235)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

I don't mind but do you realize that where as the old line (randomString.generate(2000)) would have generated a 2000 character string that this one will generate a 2000 copies of 'dummy str' or a 9*2000 = 18,000 character string? Again, not a big deal. But I Just wanted to check that this was intentional.
</blockquote></details>

Yeah, I think current implementation might be confusing, thus we can change str variable with only one character.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 361 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-KiVyJl6-DreuNxBMCWP:-KjE90cwUU_7sYkk0X4K:b-qo8w7s) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/aeda9bc19fbb3db1b55c78d84aa07db8f56023d2/test/www/jxcore/bv_tests/testThaliMobileNative.js#L361)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary><blockquote>

This still hasn't been fixed
</blockquote></details>

Typo was fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (6e08aea)

See https://github.com/ThaliTester/TestResults/tree/1159052276e08aea_Http_tests_opetkevich/ for the logs

**Comment by dersim-davaod** [comment]
Reviewed 1 of 2 files at r9, 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by dersim-davaod** [comment]
Reviewed 1 of 3 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by yaronyg** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r9, 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855#-:-KjExx-C-dOATKXb1nlr:bnfp4nl)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (f36dbe6)

**Comment by ThaliTester** [comment]
Test 115905227 (f36dbe6) build started.

**Comment by dersim-davaod** [comment]
Reviewed 1 of 1 files at r11.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1855)*

**Comment by ThaliTester** [comment]
Test (Fail) 115905227 build is completed (f36dbe6)

See https://github.com/ThaliTester/TestResults/tree/115905227f36dbe6_Http_tests_opetkevich/ for the logs
